### PR TITLE
Remove mypy skip flags

### PR DESCRIPTION
**Pull request by TechNiick**
# Summary

This PR is related to [this](https://github.com/encode/starlette/pull/2445) issue. 
This is the final PR, deleted the flags.
# Checklist

- [ ] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [ ] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [ ] I've updated the documentation accordingly.

### CODE REVIEW

File: scripts/check [hunk 1]
```diff
@@ -12,6 +12,5 @@ set -x
 ${PREFIX}ruff format --check --diff $SOURCE_FILES
 # TODO: Use `[[tool.mypy.overrides]]` on the `pyproject.toml` when the mypy issue is solved:
 # github.com/python/mypy/issues/10045. Check github.com/encode/starlette/pull/2180 for more info.
```
Comment (Kludex) [inline scripts/check:14]: This can be removed, and the part on the pyproject that is commented as well.

File: pyproject.toml [hunk 2]
```diff
@@ -66,13 +66,10 @@ python_version = "3.8"
...
 implicit_optional = true
 
-# TODO: Uncomment the following configuration when
-# https://github.com/python/mypy/issues/10045 is solved. In the meantime,
-# we are calling `mypy tests` directly. Check `scripts/check` for more info.
-# [[tool.mypy.overrides]]
-# module = "tests.*"
-# disallow_untyped_defs = false
-# check_untyped_defs = true
+ [[tool.mypy.overrides]]
```
Comment (Kludex) [inline pyproject.toml]: You can delete those.

**Review by Kludex — CHANGES_REQUESTED** [review]

**Comment by Kludex** [comment]
Thanks for your work on all the test files @TechNiick 🙏

**Review by Kludex — APPROVED** [review]

**Comment by TechNiick** [comment]
> Thanks for your work on all the test files @TechNiick 🙏

Thank you for guidance!

**Pull request merged**